### PR TITLE
chia: init at 1.1.5

### LINKED ISSUE #121492: chia-blockchain
**Project description**

Chia is a new blockchain by the creator of BitTorrent which uses proof of (storage) space to verify transactions which is a lot more economical than the ~4.5GW of GPU power the Ethereum network currently uses.

The CLI is Python and the optional GUI is Electron.

**Metadata**

```Nix
{
  pname = "chia";
  version = "1.1.3";

  src = fetchFromGitHub {
    owner = "Chia-Network";
    repo = "chia-blockchain";
    rev = version;
    sha256 = "0z6yhgjz1y6qfmbi30c64366whc8r620slw9wbn2g3d7p31x7kk9";
  };

  meta = with lib; {
    homepage = "https://www.chia.net/";
    description = "Chia is a modern cryptocurrency built from scratch, designed to be efficient, decentralized, and secure.";
    license = with licenses; [ asl20 ];
    platforms = platforms.all;
  };
}
```

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Motivation for this change
Close #121492

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/team-list.nix [hunk 1]
```diff
@@ -48,6 +48,13 @@ with lib.maintainers; {
     scope = "Maintain Cinnamon desktop environment and applications made by the LinuxMint team.";
   };
 
+  chia = {
+    members = [
+      lourkeur
```
Comment (lourkeur) [inline maintainers/team-list.nix:53]: @Atemu may I have your permission to include you in this team?

**Comment by lourkeur** [comment]
still need to:
- [ ] figure out the chia testsuite
- [ ] figure out the aarch64 build (I have a RPi4)
- [ ] lint
- [ ] set maintainer = team

File: pkgs/applications/blockchains/chia/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  ];
+
+  nativeBuildInputs = with python3Packages; [
+    setuptools_scm
+  ];
+
+  # give a hint to setuptools_scm on package version
+  preBuild = ''
+    export SETUPTOOLS_SCM_PRETEND_VERSION="v${version}"
+  '';
```
Comment (jonringer) [inline pkgs/applications/blockchains/chia/default.nix:25]: I think this looks cleaner
```suggestion
  SETUPTOOLS_SCM_PRETEND_VERSION = "v${version}";
```

File: pkgs/development/libraries/ghc_filesystem/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, lib, cmake, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "filesystem";
+  version = "1.5.4";
```
Comment (jonringer) [inline pkgs/development/libraries/ghc_filesystem/default.nix:5]:
```suggestion
  version = "1.5.4";

```

File: pkgs/development/python-modules/blspy/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        repo = "relic";
+        rev = "1885ae3b681c423c72b65ce1fe70910142cf941c"; # pinned by blspy
+        hash = "sha256-tsSZTcssl8t7Nqdex4BesgQ+ACPgTdtHnJFvS9josN0=";
+      };
+    })
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    setuptools_scm
```
Comment (jonringer) [inline pkgs/development/python-modules/blspy/default.nix:39]: please use canonical attr name
```suggestion
    setuptools-scm
```

File: pkgs/development/python-modules/chiabip158/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+buildPythonPackage rec {
+  pname = "chiabip158";
+  version = "1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-dG6A4n30uPswQWY/Wmi75HK4ZMCDNr9Lt05FRWEPYV8=";
+  };
+
+  nativeBuildInputs = [ cmake setuptools_scm ];
```
Comment (jonringer) [inline pkgs/development/python-modules/chiabip158/default.nix:20]:
```suggestion
  nativeBuildInputs = [ cmake setuptools-scm ];
```

File: pkgs/development/python-modules/chiabip158/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  buildInputs = [ pybind11 ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  disabled = pythonOlder "3.7";
+
```
Comment (jonringer) [inline pkgs/development/python-modules/chiabip158/default.nix:29]: please move to under version attr location.

File: pkgs/development/python-modules/concurrent-log-handler/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  ];
+
+  pythonImportsCheck = [
+    "concurrent_log_handler"
+  ];
+
+  doCheck = false; # upstream has no tests
+
+  meta = with lib; {
+    description = "Python logging handler that allows multiple processes to safely write to the same log file concurrently. This module extends the standard RotatingFileHandler functionality, and can be use as a drop-in replacement for that logging class.";
```
Comment (jonringer) [inline pkgs/development/python-modules/concurrent-log-handler/default.nix:27]: Description should be concise. Please use `longDescription` for lengthy descriptions.

File: pkgs/development/python-modules/keyrings-cryptfile/default.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    argon2_cffi
+    keyring
+    pycryptodome
+  ];
+
+  pythonImportsCheck = [
+    "keyrings.cryptfile"
+  ];
+
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/keyrings-cryptfile/default.nix:30]: please comment why they are disabled

File: pkgs/top-level/python-packages.nix [hunk 9]
```diff
@@ -1429,6 +1437,12 @@ in {
 
   clustershell = callPackage ../development/python-modules/clustershell { };
 
+  clvm = callPackage ../development/python-modules/clvm { };
+
+  clvm_rs = callPackage ../development/python-modules/clvm_rs { };
+
+  clvm_tools = callPackage ../development/python-modules/clvm_tools { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:1444]: Please normalize the attribute name. According to the [python contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines), package names within nix should follow [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names)
```suggestion
  clvm-rs = callPackage ../development/python-modules/clvm-rs { };

  clvm-tools = callPackage ../development/python-modules/clvm-tools { };
```

File: maintainers/team-list.nix (hunk 1 above)
Comment (Atemu) [reply maintainers/team-list.nix:53]: Sure!

Comment (lourkeur) [reply maintainers/team-list.nix:54]: Done!

**Review by Atemu — APPROVED** [review]
Looks pretty clean and well done.

Tested and it's working; the GUI isn't included but that beast can come at another time.

For completeness sake; maybe also set `meta.platform = platforms.all;` as chia should be able to run on all of them.

I'm not so sure about the cmake patches; is there not a built-in mechanism of overriding dependencies that are supposed to be fetched? They don't seem too unmaintainable though.

**Comment by lourkeur** [comment]
> For completeness sake; maybe also set meta.platform = platforms.all; as chia should be able to run on all of them.

:heavy_check_mark: 

> I'm not so sure about the cmake patches; is there not a built-in mechanism of overriding dependencies that are supposed to be fetched? They don't seem too unmaintainable though.

No. I'm not a CMake expert, but I haven't found anything from the reading I did. There is [precedent for using patches](https://github.com/NixOS/nixpkgs/blob/8284fc30c84ea47e63209d1a892aca1dfcd6bdf3/pkgs/applications/misc/crow-translate/dont-fetch-external-libs.patch) for that in the source tree, and I haven't found a tool or another way to proceed. I guess we're going to have to do it manually for now. It shouldn't break too often though.

File: pkgs/development/python-modules/blspy/dont_fetch_dependencies.patch [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+index 255e3bb..5f99c3a 100644
+--- a/python-bindings/CMakeLists.txt
++++ b/python-bindings/CMakeLists.txt
+@@ -6,8 +6,7 @@ include(FetchContent)
+ 
+ FetchContent_Declare(
+   pybind11
+-  GIT_REPOSITORY https://github.com/pybind/pybind11.git
+-  GIT_TAG        v2.6.2
++  SOURCE_DIR @pybind11_src@
```
Comment (jonringer) [inline pkgs/development/python-modules/blspy/dont_fetch_dependencies.patch:11]: these patches will probably be a pain to maintain. Probably would be a good idea to ask upstream to allow for `pkg-config` to be added to allow for easy detection of installed libraries.

However, this shouldn't block the PR.

**Review by jonringer — APPROVED** [review]
otherwise diff LGTM

**Review by jonringer — APPROVED** [review]
LGTM
shows usage

```
https://github.com/NixOS/nixpkgs/pull/122324

21 packages built:
chia ghc_filesystem python38Packages.blspy python38Packages.chiabip158 python38Packages.chiapos python38Packages.chiavdf python38Packages.clvm python38Packages.clvm-rs python38Packages.clvm-tools python38Packages.concurrent-log-handler python38Packages.keyrings-cryptfile python39Packages.blspy python39Packages.chiabip158 python39Packages.chiapos python39Packages.chiavdf python39Packages.clvm python39Packages.clvm-rs python39Packages.clvm-tools python39Packages.concurrent-log-handler python39Packages.keyrings-cryptfile tests.trivial
```

**Pull request merged**